src/dhcp_mon.cpp

```diff
@@ -318,13 +485,26 @@ int dhcp_mon_start(size_t snaplen, bool debug_mode)
             break;
         }
 
+        struct event *ev_sigusr2 = evsignal_new(main_event_mgr->get_base(), SIGUSR2, signal_callback, main_event_mgr->get_base());
+        if (ev_sigusr2 == NULL) {
+            syslog(LOG_ERR, "Could not create SIGUSR2 libevent signal!\n");
+            break;
+        }
+
+    
+        cache_db_diff = false;
+        dhcp_packet_direction_t temp_dir_rx = DHCP_RX;
+        ev_rx_cache_counter_update = event_new(rx_event_mgr->get_base(), -1, 0, update_cache_counter, reinterpret_cast<void*>(&temp_dir_rx));
+        dhcp_packet_direction_t temp_dir_tx = DHCP_TX;
+        ev_tx_cache_counter_update = event_new(tx_event_mgr->get_base(), -1, 0, update_cache_counter, reinterpret_cast<void*>(&temp_dir_tx));
+
         struct event *ev_timeout = event_new(main_event_mgr->get_base(), -1, EV_PERSIST, timeout_callback, main_event_mgr->get_base());
         if (ev_timeout == NULL) {
             syslog(LOG_ERR, "Could not create libevent timer!\n");
             break;
         }
 
-        struct event *ev_db_update = event_new(main_event_mgr->get_base(), -1, EV_PERSIST, db_update_callback, main_event_mgr->get_base());
+        ev_db_update = event_new(main_event_mgr->get_base(), -1, EV_PERSIST, db_update_callback, main_event_mgr->get_base());
```

Comment:
Do you want to ensure the `ev_db_update` is NULL before assign a new to it? 
An assertion maybe more reasonable if you expect it never be NULL.

Reply:
Add del and free event if it's not null